manual/thread_safetiness/index.md

```diff
@@ -0,0 +1,23 @@
+---
+title: Thread safetiness
+layout: single
+sidebar:
+  nav: "manual"
+toc: true
+toc_sticky: true
+---
+
+ROOT classes support one of the following thread-safety levels:
+
+- Thread unsafe: objects must not be used concurrently by different threads, even if objects are not shared among threads.
+
+- Conditionally thread safe: objects can be used safely by multiple threads concurrently as long as mutiple threads do not share the same object (i.e. every thread uses a private copy of its objects).  In addition, objects can be shared among threads as long as all threads use only `const` methods of the shared objects.
```

Comment:
A suggestion that's just to hint at what I think should be made clearer, not meant to be accepted as is:

```suggestion
- Thread-local instances are safe: different threads can safely use different instances of the given type (i.e. every thread uses a local instance of this type).  In addition, objects can be shared among threads as long as all threads use only `const` methods of the shared objects. Instances of different types can freely be used concurrently.
```